runtime/near-vm-runner/src/errors.rs

```diff
@@ -65,10 +65,62 @@ impl IntoVMError for wasmer_runtime::error::ResolveError {
 
 impl IntoVMError for wasmer_runtime::error::RuntimeError {
     fn into_vm_error(self) -> VMError {
+        use near_vm_errors::WasmTrap::{BreakpointTrap, GenericTrap};
+        use wasmer_runtime::error::InvokeError;
         use wasmer_runtime::error::RuntimeError;
         match &self {
-            RuntimeError::Trap { msg: _ } => VMError::FunctionCallError(WasmUnknownError),
-            RuntimeError::Error { data } => {
+            RuntimeError::InvokeError(invoke_error) => match invoke_error {
+                // Indicates an exceptional circumstance such as a bug in Wasmer
+                // or a hardware failure.
+                // As of 0.17.0, thrown when stack unwinder fails, or when
+                // invoke returns false and doesn't fill error info,
+                // what Singlepass BE doesn't.
+                InvokeError::FailedWithNoError => {
+                    panic!("Impossible FailedWithNoError error");
+                },
+                // Indicates that a trap occurred that is not known to Wasmer.
+                // As of 0.17.0, thrown only from Cranelift BE.
+                InvokeError::UnknownTrap { address: _, signal: _ } => {
+                    panic!("Impossible UnknownTrap error");
+                }
+                // A trap that Wasmer knows about occurred.
+                // As of 0.17.0, thrown only from LLVM and Cranelift BE.
+                InvokeError::TrapCode { code: _, srcloc: _ } => {
+                    panic!("Impossible TrapCode error");
+                }
+                // A trap occurred that Wasmer knows about but it had a trap code that
+                // we weren't expecting or that we do not handle.
+                // As of 0.17.0, thrown only from Cranelift BE.
+                InvokeError::UnknownTrapCode { trap_code: _, srcloc: _ } => {
+                    panic!("Impossible UnknownTrapCode error");
```

Comment:
Would be helpful to add `trap_code` and `srcloc` to the panic message, just in case we mess up and have cranelift or llvm running. Same for other panics above.